packages/vm/src/context.rs

```diff
@@ -250,21 +250,24 @@ mod iter_support {
             // But there is no way for the compiler to know. So... let's just lie to the compiler a little bit.
             let live_forever: Box<dyn Iterator<Item = StdResult<KV>> + 'static> =
                 unsafe { mem::transmute(iter) };
-            let new_id = set_iterator::<S, Q>(ctx, live_forever);
-            Ok(match to_i32(new_id) {
-                Ok(new_id_signed) => new_id_signed,
-                Err(_) => ERROR_DB_UNKNOWN,
-            })
+            Ok(live_forever)
         });
-        match res {
-            Ok(callback_result) => callback_result,
+
+        match range_result {
+            Ok(iterator) => {
+                let new_id = set_iterator::<S, Q>(ctx, iterator);
```

Comment:
nice to move this out of above closure

Reply:
This was even necessary, since `set_iterator` and `with_storage_from_context` now both need a mutable reference to Ctx. The refactoring revealed that.